js/react-native-source-maps/index.js

```diff
@@ -29,6 +32,37 @@ export const initSourceMaps = async opts => {
   options = opts
 }
 
+const createSourceMapper = async () => {
+  const bundlePath =
+    Platform.OS === 'ios' ? RNFS.MainBundlePath : RNFS.DocumentDirectoryPath
+  const path = `${bundlePath}/${options.sourceMapBundle}`
+  try {
+    const fileExists = await RNFS.exists(path)
+    if (!fileExists) {
+      throw new Error(
+        __DEV__
+          ? 'Unable to read source maps in DEV mode'
+          : `Unable to read source maps, possibly invalid sourceMapBundle file, please check that it exists here: ${
+              bundlePath
+            }/${options.sourceMapBundle}`
+      )
+    }
+
+    const mapContents = await RNFS.readFile(path, 'utf8')
+    const sourceMaps = JSON.parse(mapContents)
+    const mapConsumer = new SourceMap.SourceMapConsumer(sourceMaps)
+
+    return (sourceMapper = row => {
+      return mapConsumer.originalPositionFor({
+        line: row.lineNumber,
+        column: row.columnNumber
+      })
+    })
+  } catch (error) {
```

Comment:
uhmmm :) this is like not having a try/catch block, right?

Reply:
The function is async, i think if an exception is raised for the internal Promise you need to rethrow it or the main function caller will not receive the exception.